io-sim/src/Control/Monad/IOSimPOR/Internal.hs

```diff
@@ -1623,6 +1598,15 @@ execWriteTVar :: TVar s a -> a -> ST s ()
 execWriteTVar TVar{tvarCurrent} = writeSTRef tvarCurrent
 {-# INLINE execWriteTVar #-}
 
+execTryPutTMVar :: TVar s (Maybe a) -> a -> ST s Bool
+execTryPutTMVar var a = do
```

Comment:
This function is called `execTryPutTMVar` but receives a `TVar`

Reply:
The alternative is:
```
execTryPutTMVar :: TMVar (IOSim s) a -> a -> ST s Bool
```

Reply:
I changed to `TMVar`s.